.pre-commit-config.yaml

```diff
@@ -0,0 +1,16 @@
+repos:
+  -   repo: https://github.com/timothycrosley/isort
+      rev: 5.6.4
```

Comment:
Anyway not to hardcode version? We have CI always use the latest. It'd be a bit annoying to have to bump them here, I think.

Reply:
To quote the relevant extract from the pre-commit docs. 

_therefore intentionally doesn't provide facilities for "unpinned latest version" for hook repositories.

Instead, pre-commit provides tools to make it easy to upgrade to the latest versions with pre-commit autoupdate_

https://pre-commit.com/#using-the-latest-version-for-a-repository

Reply:
Using a setup similar to that suggested by Adam would avoid this. But would then need to explain how to setup tox. I was concerned about this as I thought it would be too much for the introduction docs. But if it's targetted more at regular contributors then this could be ok? 

https://github.com/adamchainz/apig-wsgi/blob/master/.pre-commit-config.yaml

Reply:
Bumping versions so frequently will be really annoying :disappointed: :

- 2020 (35 `isort` releases + 5 `flake8` releases = 40 :exclamation:)
- 2019 (17 `isort` releases + 10 `flake8` releases = 27)
- ...

Reply:
We don't need every release. What I do on my personal repos is pin everything for reproducible builds and upgrade all requirements every two weeks with a script (e.g. https://github.com/adamchainz/django-mysql/pull/727 ).

We could run `pre-commit autoupdate` every two weeks on django. I don't think it's a lot of work and I'm happy to take that on if no one else is.

Reply:
We don't pin `isort` and `flake8` versions in contributing docs, tests requirements, CI, tox configuration etc. so it will be inconsistent. I can imagine that `pre-commit` hooks will pass but CI will fail :thinking: I'm a bit pessimistic about `pre-commit`, so I'm going to leave this patch for Carlton, who likes it :wink:

Reply:
Thanks @felixxm 😅

Reply:
> We don't pin isort and flake8 versions in contributing docs, tests requirements, CI, tox configuration etc. 

Just a thought, but we could route all those instances through pre-commit for consistent, reproducible linting. To run all pre-commit tools on the full project:

```
pre-commit run --all-files
```

The `--show-diff-on-failure` is also helpful for debugging.

Reply:
You could use pre-commit.ci to autoupdate your pre-commit config